src/telemetry/telemetry.c

```diff
@@ -0,0 +1,290 @@
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <postgres.h>
+#include <access/xact.h>
+#include <fmgr.h>
+#include <miscadmin.h>
+#include <commands/extension.h>
+#include <utils/builtins.h>
+#include <utils/json.h>
+#include <utils/jsonb.h>
+
+#include "compat.h"
+#include "guc.h"
+#include "telemetry.h"
+#include "metadata.h"
+#include "hypertable.h"
+#include "net/utils.h"
+#include "net/uri.h"
+
+#ifndef WIN32
+#include <sys/utsname.h>
+#endif
+
+#define TS_VERSION_JSON_FIELD "current_timescaledb_version"
+
+/*  HTTP request details */
+#define TIMESCALE_URI	"/v1/metrics"
+#define TIMESCALE_TYPE	"application/json"
+#define MAX_REQUEST_SIZE	4096
+
+#define REQ_DB_UUID					"db_uuid"
+#define REQ_EXPORTED_DB_UUID		"exported_db_uuid"
+#define REQ_INSTALL_TIME			"installed_time"
+#define REQ_INSTALL_METHOD			"install_method"
+#define REQ_OS						"os_name"
+#define REQ_OS_VERSION				"os_version"
+#define REQ_OS_RELEASE				"os_release"
+#define REQ_PS_VERSION				"postgresql_version"
+#define REQ_TS_VERSION				"timescaledb_version"
+#define REQ_BUILD_OS				"build_os_name"
+#define REQ_BUILD_OS_VERSION		"build_os_version"
+#define REQ_DATA_VOLUME				"data_volume"
+#define REQ_NUM_HYPERTABLES			"num_hypertables"
+#define REQ_RELATED_EXTENSIONS		"related_extensions"
+
+#define PG_PROMETHEUS	"pg_prometheus"
+#define POSTGIS			"postgis"
+
+static const char *related_extensions[] = {PG_PROMETHEUS, POSTGIS};
+static const char *version_delimiter[3] = {".", ".", ""};
+
+/*
+ * Parse the JSON response from the TS endpoint. There should be a field
+ * called "current_timescaledb_version". Check this against the local
+ * version, and notify the user if it is behind.
+ */
+static void
+process_response(const char *endpoint_response)
```

Comment:
Could you wrap this in a tester function that enables us to submit a variety (including ill-formed?) `endpoint_response` strings in `test/sql/telemetry.sql`?